nox.py

```diff
@@ -86,6 +86,11 @@ def session_tests(session, interpreter, extra_pytest_args=None):
     # allows users to run a particular test instead of all of them.
     for sample in (session.posargs or
                    collect_sample_dirs('.', SESSION_TESTS_BLACKLIST)):
+        # Install additional dependencies if they exist
```

Comment:
This could cause some weirdness with clashing dependency versions. I'm okay with this for now, but we should be careful.